### PR TITLE
Feat Add tracking to save star press event

### PR DESCRIPTION
**Pull request by GaboDimitrov**
Add tracking to save star press event. Only articleId is going to be tracked. Save star tests are refactored.

### CODE REVIEW

File: packages/save-and-share-bar/src/save-and-share-bar.js [hunk 1]
```diff
@@ -25,6 +26,37 @@ class SaveAndShareBar extends Component {
     onCopyLink();
   }
 
+
+  onSaveButtonPress(evt) {
```
Comment (nehasri89) [inline packages/save-and-share-bar/src/save-and-share-bar.js]: why do we need to copy code from `save-star`, we could have done the tracking in `save-star`

File: packages/save-star-web/__tests__/shared.base.js [hunk 2]
```diff
@@ -1,212 +1,51 @@
...
-        const apiMock = {
-          ...mockSaveApi,
-          getBookmarks: () =>
-            Promise.resolve({
-              loading: true
-            })
-        };
-
-        const testInstance = TestRenderer.create(
-          <SaveStarWeb
```
Comment (nehasri89) [inline packages/save-star-web/__tests__/shared.base.js]: these scenarios need to be tested, I don't see these being added anywhere else

Comment (ValGeorgiev) [reply packages/save-star-web/__tests__/shared.base.js]: They are obsolete. As we discuss with Nate, in the first PR, this testing of state is not quite right. We have test for save, unsave and loading scenario. I think that is enough. What you think?

File: packages/save-and-share-bar/src/save-and-share-bar.js (hunk 1 above)
Comment (ValGeorgiev) [reply packages/save-and-share-bar/src/save-and-share-bar.js]: the tracking is done in SaveStar, could you suggest how to change the tracking to not have onSaveButtonPress in save-and-share-bar ? The tracking looks to works in that way

File: packages/save-star-web/__tests__/shared.base.js (hunk 2 above)
Comment (nehasri89) [reply packages/save-star-web/__tests__/shared.base.js]: okay, I understand your point. There was a bug in where the incorrect apis where being invoked, for which I have put few tests. Those tests are pretty imp I think

File: packages/save-star-web/src/save-star-web.js [hunk 3]
```diff
@@ -59,14 +52,13 @@ class SaveStarWeb extends Component {
       })
       .catch(error => {
         this.setState({ loadingState: false, savedStatus: false });
-        console.error("Error in connecting to api", error);
+        console.error(error);
```
Comment (IvayloAndonov) [inline packages/save-star-web/src/save-star-web.js]: _Not related with the PR_

**Disclaimer**: The 'console' is useful only for testing purposes. Has our webpack been configured to remove all 'console' logs across the code on production env ?  Better approach is to use some logger here.

Comment (ValGeorgiev) [reply packages/save-star-web/src/save-star-web.js]: @IvayloAndonov this is very useful ^, could you raise it on the next tech dept session, please?

**Pull request merged**